src/sinks/util/adaptive_concurrency/controller.rs

```diff
@@ -243,7 +243,12 @@ where
                 } else if error.downcast_ref::<Elapsed>().is_some() {
                     true
                 } else {
-                    unreachable!("Unhandled error response! {:?}", error)
+                    warn!(
```

Comment:
Would it make sense to emit an event here for this?

Reply:
I don't think so, as it's more of a programming error than an operational error, but I could be convinced otherwise.

Reply:
Gotcha, I see. Maybe we could print a message asking them to file a bug then? I've seen that for some other programs where "unreachable" code is reached.

Reply:
@jszwedko this is blocking 0.11. Can we defer this and open an issue?

Reply:
That's fine with me :+1: